shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -690,6 +691,14 @@ private void flushAllViews() {
       controller.dispose();
     }
     vdControllers.clear();
+
+    int[] viewIds = new int[platformViews.size()];
+    for (int i = 0; i < platformViews.size(); i++) {
+      viewIds[i] = platformViews.keyAt(i);
+    }
+    for (int viewId : viewIds) {
+      channelHandler.disposeAndroidViewForPlatformView(viewId);
```

Comment:
could it just be a single loop?

```java
for (int i = 0; i < platformViews.size(); i++) {
  final int viewId = platformViews.keyAt(i);
  channelHandler.disposeAndroidViewForPlatformView(viewId);
}
```

Reply:
That won't work because the indexes of the keys will change each time `disposeAndroidViewForPlatformView` removes a key from `platformViews`.

Changed this to:
```
while (platformViews.size())
  remove(platformViews.keyAt(0))
```